src/main/kotlin/com/hypto/iam/server/authProviders/WorkOSAuthProvider.kt

```diff
@@ -0,0 +1,31 @@
+package com.hypto.iam.server.authProviders
+
+import com.hypto.iam.server.ROOT_ORG
+import com.hypto.iam.server.configs.AppConfig
+import com.hypto.iam.server.security.AuthMetadata
+import com.hypto.iam.server.security.OAuthUserPrincipal
+import com.hypto.iam.server.security.TokenCredential
+import com.workos.WorkOS
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+object WorkOSAuthProvider : BaseAuthProvider("workos"), KoinComponent {
+    private val appConfig: AppConfig by inject()
+
+    override fun getProfileDetails(tokenCredential: TokenCredential): OAuthUserPrincipal {
+        val workos = WorkOS(appConfig.workOS.secretKey)
```

Comment:
Instead of creating new WorkOS object for every request, can we create a single object and inject it at object level?

Reply:
made singleton